lib/rules/react-hooks-strict-return.js

```diff
@@ -0,0 +1,154 @@
+const {docsUrl} = require('../utilities');
+
+const MAX_RETURN_ELEMENTS = 2;
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Restrict the number of returned items from React hooks.',
+      category: 'Best Practices',
+      recommended: true,
+      uri: docsUrl('react-hooks-strict-return'),
+    },
+    messages: {
+      hooksStrictReturn:
+        'React hooks must return a tuple of two or fewer values or a single object.',
+    },
+  },
+
+  create(context) {
+    let inHook = false;
+
+    return {
+      VariableDeclarator(node) {
+        if (!isHook(node)) {
+          return;
+        }
+
+        inHook = true;
+      },
+      'VariableDeclarator:exit': function(node) {
+        if (!isHook(node)) {
+          return;
+        }
+
+        inHook = false;
+      },
+      FunctionDeclaration(node) {
+        if (!isHook(node)) {
+          return;
+        }
+
+        inHook = true;
+      },
+      'FunctionDeclaration:exit': function(node) {
+        if (!isHook(node)) {
+          return;
+        }
+
+        inHook = false;
```

Comment:
I suspect something like this will cause the rule to not complain, even though it should:

```tsx
function useFoo() {
  useEffect(() => {});

  return [1, 2, 3, 4];
}
```

Reply:
Added some more tests to cover this https://github.com/Shopify/eslint-plugin-shopify/pull/237/files#diff-cde867597099df6969da77e4a149d359R206